source/io/module_javaScriptInvoke.js

```diff
@@ -496,13 +524,16 @@
             var returnValue = undefined;
             var api;
             if (functionToCall.length === parameters.length + 1) {
-                api = generateAPI(occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus);
+                api = generateAPI(occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus, isInternalFunction);
                 parameters.push(api);
             }
 
             try {
                 returnValue = functionToCall.apply(context, parameters);
             } catch (ex) {
+                if (isInternalFunction) {
```

Comment:
Why does this throw again if it is an internal function?

Reply:
In user JS code we don't want to crash the runtime. It should be caught and reported as a JS error back on the wire.

Our internal functions should allow the exception through because it means there is a failure in code we have written. If internal functions need to report an error on the wire they should use the jsapi for setting LabVIEW style errors, not throwing an exception. As such exceptions are actually exceptions and need to propagate upwards to fail tests or cause the editor to receive an error, etc.